### DISCUSSION TITLE
Note about Optimizations

### DISCUSSION THREAD

**Discussion by regiellis** [TheR1D/shell_gpt#574] [Q&A] [answered]
Hi,

This line appears in the docs twice **"Note that ShellGPT is not optimized for local models and may not work as expected."** with
no explanation on what "optimizations" is happening for OpenAI models vs local models. Are there feature sets missing? Is it
just slower?, limited feedback?

It would be helpful to know when making the choice on what setup to use...

**Comment by TheR1D** [comment, marked as answer]
ShellGPT was primarily developed to target the OpenAI API, with functionality tested exclusively with OpenAI’s language models. For instance, some local language models might lack the function calling feature or may not be compatible with OpenAI’s [function calls](https://platform.openai.com/docs/guides/function-calling). Additionally, certain local models might completely ignore “system” prompts like "provide only shell commands without any description". As a result, when a user attempts to generate a shell command using the `--shell` option, a local model might format the output in markdown, add a description, or introduce other elements, rather than providing a clear, executable shell command. 

I agree that this description is not very clear. We will add more details to make it more understandable.
